lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb

```diff
@@ -198,8 +198,13 @@ def line_item_includes
       private
 
       def shipping_method(line_items)
-        line_items.first.order.shipments.first.
-          andand.shipping_rates.andand.first.andand.shipping_method
+        shipping_rates = line_items.first.order.shipments.first.
+          andand.shipping_rates
+
+        return unless shipping_rates
+
+        shipping_rate = shipping_rates.find(&:selected) || shipping_rates.first
+        shipping_rate.try(:shipping_method)
```

Comment:
Is this repetition really needed? if it is it should be merged with already existing methods in OrderShipment (shipment and shipping_method) and Shipment (selected_shipping_rate)

Reply:
I am only making this point here because this is copy/pasted core business logic. If it was some report specific logic I wouldn't worry.

Reply:
I initially tried to adapt the related method directly, but there were various spec failures in Subs and order-related services. I'm not sure how long it would take to get a green build going in that direction; it's preferable but might not be  quick.

Reply:
I didn't have much time and I felt the urgency around unblocking the release, but maybe the sensible option is to revert for now and hold back the reports improvements until the next release?

Reply:
that's one option as well. I think we can go either way: revert to original state or keep this PR as is, as long as we improve it in the next release.

Do you want to share a PR with the change of the methods that break the specs, maybe I can help fix it?

Reply:
the "loaded?" approach didn't work?

Reply:
No, I this this again (in the `Spree::Shipment` decorator), which I think is the way to go:

```ruby
def selected_shipping_rate
      if shipping_rates.loaded?
        shipping_rates.find(&:selected)
      else
        shipping_rates.where(selected: true).first
     end
end
```

but there were multiple failures here (I think Job #3 is just a flaky one): https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/pull-request-4469/builds/2

Reply:
ok, I'll have a look at that!